### DISCUSSION TITLE
Client ID/map node ID/whichever

### DISCUSSION THREAD

**Discussion by Miss-Inputs** [sk-zk/streetlevel#5] [Ideas]
Looking into it a bit, of course I cannot know for sure but I am sure it has _something_ to do with the client ID, from the pano lw7Qe7mxk3z3Ck-r5p0X3w we get a reference to the place the trekker is of (apologies if this should have been another test case):
```
               [
                  [
                     [
                        null,
                        [
                           "7716440184157373679",
                           "4751164214735023334"
                        ],
                        null,
                        "4751164214735023334"
                     ],
                     null,
                     [
                        "Australian War Memorial",
                        "en"
                     ],
                     [
                        "War memorial",
                        "en"
                     ],
                     "https://maps.gstatic.com/mapfiles/annotations/icons/generic_2x.5.png",
                     null,
                     null,
                     2
                  ],
                  [
                     [
                        [
                           "/m/028vq2"
                        ]
                     ]
                  ]
               ]
```

This all matches up with what I am told about feature IDs/CIDs/place IDs from here: https://outscraper.com/place-id-feature-id-cid/, which may be completely wrong, because at the end of the day they are also guessing what things do or mean (and are also trying to sell some kind of weird SEO service, but those kind of results are all you get for trying to search for "google maps cid" on the internet), and who can really know for sure except Google employees?

If we visit `https://maps.google.com/?cid=4751164214735023334` we are taken there. This is equal to the `url` field of the Place object in the official API (last time I checked anyway with the developer docs examples). So, assuming that the terminology is consistent, that place[0][3] field (which is also equal to place[0][1][1], as is expected from a feature ID) is the CID. The URL we get redirected to also has 0x6b164d76cb4004ef:0x41ef86ae38d42ce6 in there, which is the hex encoded ID from places[0][1] as expected, and looks like what those random SEO sites call a "feature ID".

Links like those don't work for "Geocoded address" or "Intersection" or anything else without a name, as mentioned in the comments for the places PR, those are weird and behave differently.

~~I had a shower thought that the `/m/028vq2` may be related to the Arts & Culture stuff. `028vq2` is not the asset ID for the Street View for the Australian War Memorial (that is `QgHJq7wlm9eraA`), but it seems that these one-element objects that get returned in the places array always (so far) start with /g/ and /m/ - could be gallery or museum? Could also just not be.~~ Nope, appears for some other random trekkers.

There is also sometimes that 9-element array which contains 8 nulls and something like `SV4Rn9v-JZU` (as seen in the Old Parliament House test case), which I am sure is some kind of ID, but who knows what.

Was really hoping I could get the place ID `ChIJ7wRAy3ZNFmsR5izUOK6G70E` in the response, but no such luck. Looking at the requests from Google Maps does show calls to `https://www.google.com/maps/preview/place`, which returns some stuff that has the place ID and other nice info in there. I don't think the internal places API is in scope for this project though.

**Comment by sk-zk** [comment]
Good to know, thanks. I also found this:
https://dataforseo.com/help-center/what-is-cid-place-id-feature-id
so this is all an established thing I guess. I'm gonna uncomment the field (but change the name to `cid` because none of these pages agree on what it stands for)

As for getting a place ID from a cid, there are a few ways:
https://stackoverflow.com/questions/47490721/from-google-maps-cid-to-place-id
But yeah, that's out of scope for this project.

**Comment by sk-zk** [comment]
Just found out what the the /m/ and /g/ entries are: they're IDs for Google Knowledge Graph. The /m/ IDs are imported from Freebase, and the /g/ IDs are unique to Google's database.
If you google `/m/028vq2`, you'll find the Wikidata page for the Australian War Memorial which lists this ID: https://www.wikidata.org/wiki/Q782783
